lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -1191,6 +1206,56 @@ private Task claim(String taskId, boolean forceClaim)
     return task;
   }
 
+  private Task requestReview(String taskId, boolean force)
+      throws TaskNotFoundException, NotAuthorizedException, InvalidTaskStateException,
+          InvalidOwnerException {
+    String userId = taskanaEngine.getEngine().getCurrentUserContext().getUserid();
+    TaskImpl task;
+    try {
+      taskanaEngine.openConnection();
+      task = (TaskImpl) getTask(taskId);
+
+      TaskImpl oldTask = task.copy();
+      oldTask.setId(taskId);
+      oldTask.setExternalId(task.getExternalId());
+      oldTask.setAttachments(task.getAttachments());
+      oldTask.setSecondaryObjectReferences(task.getSecondaryObjectReferences());
```

Comment:
we've seen this pattern so often in this class. Do you mind extracting this into a dedicated method (naming could be something like duplicateTaskExactly) and replace all usages of `copy` within this class to use that method.
I just saw that e.g. the claim method has some errors during the copying since it only sets the id and externalId after the `copy` command. With that method we have the logic for duplicating a task only in one place :)

Reply:
Ok :)